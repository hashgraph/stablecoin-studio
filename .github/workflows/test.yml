name: test
on:
  pull_request:
    branches: [ main, release/**, feature/** ]
  push:
    branches: [ main, release/** ]
    tags: [ v* ]
jobs:
  test-node:
    name: Unit Tests
    runs-on: ubuntu-latest
    steps:
      - name: Use Node.js TLS 16.x
        uses: actions/setup-node@v1
        with:
          node-version: 16.x
      - name: Checkout repository
        uses: actions/checkout@v2
      - name: Install dependencies
        run: npm i
      - name: Build
        run: npm run build
      - name: Test
        run: npm run test