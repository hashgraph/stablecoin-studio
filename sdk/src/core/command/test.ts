/* eslint-disable @typescript-eslint/no-empty-interface */

interface CommandResponse {}
interface ICommand {}

export class Command<T> implements ICommand {
	private $resultType!: T;
}

export interface ICommandBus {
	execute<X>(command: Command<X>): Promise<X>;
}

export type IInferringCommandHandler<CommandType extends Command<unknown>> =
	CommandType extends Command<infer ResultType>
		? {
				execute(command: CommandType): Promise<ResultType>;
		  }
		: never;

class ConcreteCommandResponse implements CommandResponse {
	constructor(public readonly status: string) {}
}

class ConcreteCommand extends Command<ConcreteCommandResponse> {
	constructor(public readonly value: string) {
		super();
	}
}

class CommandHandler<CommandType extends Command<unknown>>
	implements IInferringCommandHandler<ConcreteCommand>
{
	execute(command: ConcreteCommand): Promise<ConcreteCommandResponse> {
		throw new Error('Method not implemented.');
	}
}

class CommandBus implements ICommandBus {
	execute<X>(command: Command<X>): Promise<X> {
		throw new Error('Method not implemented.');
	}
}

// Test

const bus = new CommandBus();
const res = bus.execute(new ConcreteCommand('1'));
