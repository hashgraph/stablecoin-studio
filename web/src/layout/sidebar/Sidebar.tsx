import { Flex } from '@chakra-ui/react';
import { useTranslation } from 'react-i18next';
import { NamedRoutes } from '../../Router/NamedRoutes';
import SidebarOption from './SidebarOption';
import { SELECTED_WALLET_COIN } from '../../store/slices/walletSlice';
import { useSelector } from 'react-redux';

interface optionsProps {
	icon: string;
	title: string;
	route: NamedRoutes;
	isDisabled?: boolean;
	isHidden?: boolean;
}

const Sidebar = () => {
	const { t } = useTranslation('global');
	const selectedStableCoin = useSelector(SELECTED_WALLET_COIN);
	const options: optionsProps[] = [
		{
			icon: 'Coin',
			title: t('sidebar.operations'),
			route: NamedRoutes.Operations,
		},
		{
			icon: 'Users',
			title: t('sidebar.role'),
			route: NamedRoutes.Roles,
			isDisabled: selectedStableCoin?.deleted ?? false,
		},
		{
			icon: 'Fingerprint',
			title: t('sidebar.details'),
			route: NamedRoutes.StableCoinDetails,
		},
		{
			icon: 'Receipt',
			title: t('sidebar.feesManagement'),
			route: NamedRoutes.FeesManagement,
			isHidden: selectedStableCoin && !selectedStableCoin.feeScheduleKey,
		},
		{
			icon: 'File',
			title: t('sidebar.proofOfReserve'),
			route: NamedRoutes.ProofOfReserve,
			isHidden: selectedStableCoin && !selectedStableCoin.reserveAddress,
		},
		{
			icon: 'GearSix',
			title: t('sidebar.settings'),
			route: NamedRoutes.Settings,
		},
	];

	return (
		<Flex
			data-testid='sidebar'
			bgColor='brand.white'
			minW='240px'
			p='32px 12px'
			justifyContent={'space-between'}
			flexDirection='column'
		>
			<Flex flexDirection='column' alignItems='center' gap={3}>
				{options.map((option) => {
					return <SidebarOption key={option.title} {...option} />;
				})}
			</Flex>
		</Flex>
	);
};

export default Sidebar;
