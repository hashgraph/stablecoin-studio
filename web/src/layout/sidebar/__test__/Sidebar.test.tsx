import Sidebar from '../Sidebar';
import { render } from '../../../test';

describe(`<${Sidebar.name} />`, () => {
	test('should render correctly', () => {
		const component = render(<Sidebar />);

		expect(component.asFragment()).toMatchSnapshot();
	});

	test('should render options', () => {
		const component = render(<Sidebar />);
		const options = ['Coin', 'Users', 'Gear'];

		options.forEach((option) => {
			const element = component.getByTestId(`sidebar-option-${option}`);
			expect(element).toBeInTheDocument();
		});
	});
});
