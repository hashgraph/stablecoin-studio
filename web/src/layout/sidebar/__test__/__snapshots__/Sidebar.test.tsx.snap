// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidebar /> should render correctly 1`] = `
<DocumentFragment>
  <div
    class="css-1nfhpzz"
    data-testid="sidebar"
  >
    <div
      class="css-17lnxuq"
    >
      <button
        class="chakra-button css-9d0zi7"
        data-testid="sidebar-option-Coin"
        type="button"
      >
        <svg
          class="chakra-icon css-wrbp2a"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="0 0 256 256"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="none"
            height="256"
            width="256"
          />
          <ellipse
            cx="128"
            cy="104"
            fill="none"
            rx="104"
            ry="48"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
          />
          <line
            fill="none"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
            x1="128"
            x2="128"
            y1="152"
            y2="200"
          />
          <path
            d="M24,104v48c0,24,40,48,104,48s104-24,104-48V104"
            fill="none"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
          />
          <line
            fill="none"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
            x1="192"
            x2="192"
            y1="142.1"
            y2="190.1"
          />
          <line
            fill="none"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
            x1="64"
            x2="64"
            y1="142.1"
            y2="190.1"
          />
        </svg>
        <p
          class="chakra-text css-1d26k2o"
          data-testid="sidebar-option-title"
        >
          Operations
        </p>
      </button>
      <button
        class="chakra-button css-9d0zi7"
        data-testid="sidebar-option-Users"
        type="button"
      >
        <svg
          class="chakra-icon css-wrbp2a"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="0 0 256 256"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="none"
            height="256"
            width="256"
          />
          <circle
            cx="88"
            cy="108"
            fill="none"
            r="52"
            stroke="currentColor"
            stroke-miterlimit="10"
            stroke-width="16"
          />
          <path
            d="M155.4,57.9A54.5,54.5,0,0,1,169.5,56a52,52,0,0,1,0,104"
            fill="none"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
          />
          <path
            d="M16,197.4a88,88,0,0,1,144,0"
            fill="none"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
          />
          <path
            d="M169.5,160a87.9,87.9,0,0,1,72,37.4"
            fill="none"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
          />
        </svg>
        <p
          class="chakra-text css-1d26k2o"
          data-testid="sidebar-option-title"
        >
          Role Management
        </p>
      </button>
      <button
        class="chakra-button css-9d0zi7"
        data-testid="sidebar-option-Fingerprint"
        type="button"
      >
        <svg
          class="chakra-icon css-wrbp2a"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="0 0 256 256"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="none"
            height="256"
            width="256"
          />
          <path
            d="M50.7,184.9A127.4,127.4,0,0,0,64,128,64.2,64.2,0,0,1,88,78"
            fill="none"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
          />
          <path
            d="M128,128a191.2,191.2,0,0,1-24,93"
            fill="none"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
          />
          <path
            d="M96,128a32,32,0,0,1,64,0,222.3,222.3,0,0,1-21.3,95.4"
            fill="none"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
          />
          <path
            d="M218.6,184a294.6,294.6,0,0,0,5.4-56,96,96,0,0,0-192,0,94.4,94.4,0,0,1-5.5,32"
            fill="none"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
          />
          <path
            d="M92.8,160a161.9,161.9,0,0,1-18.1,47.9"
            fill="none"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
          />
          <path
            d="M120,64.5a70.1,70.1,0,0,1,8-.5,64,64,0,0,1,64,64,260.6,260.6,0,0,1-2,32"
            fill="none"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
          />
          <path
            d="M183.9,192c-1.5,5.9-3.2,11.8-5.1,17.5"
            fill="none"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
          />
        </svg>
        <p
          class="chakra-text css-1d26k2o"
          data-testid="sidebar-option-title"
        >
          Stablecoin Details
        </p>
      </button>
      <button
        class="chakra-button css-9d0zi7"
        data-testid="sidebar-option-GearSix"
        type="button"
      >
        <svg
          class="chakra-icon css-wrbp2a"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="0 0 256 256"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="none"
            height="256"
            width="256"
          />
          <circle
            cx="128"
            cy="128"
            fill="none"
            r="48"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
          />
          <path
            d="M197.4,80.7a73.6,73.6,0,0,1,6.3,10.9L229.6,106a102,102,0,0,1,.1,44l-26,14.4a73.6,73.6,0,0,1-6.3,10.9l.5,29.7a104,104,0,0,1-38.1,22.1l-25.5-15.3a88.3,88.3,0,0,1-12.6,0L96.3,227a102.6,102.6,0,0,1-38.2-22l.5-29.6a80.1,80.1,0,0,1-6.3-11L26.4,150a102,102,0,0,1-.1-44l26-14.4a73.6,73.6,0,0,1,6.3-10.9L58.1,51A104,104,0,0,1,96.2,28.9l25.5,15.3a88.3,88.3,0,0,1,12.6,0L159.7,29a102.6,102.6,0,0,1,38.2,22Z"
            fill="none"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="16"
          />
        </svg>
        <p
          class="chakra-text css-1d26k2o"
          data-testid="sidebar-option-title"
        >
          Settings
        </p>
      </button>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
